pyboy/core/ram.py

```diff
@@ -15,17 +16,20 @@
 
 
 class RAM:
-    def __init__(self, random=False):
-        if random: # NOTE: In real life, the RAM is scrambled with random data on boot.
-            raise Exception("Random RAM not implemented")
-
+    def __init__(self, randomize=False):
         self.internal_ram0 = array.array("B", [0] * (INTERNAL_RAM0))
         self.non_io_internal_ram0 = array.array("B", [0] * (NON_IO_INTERNAL_RAM0))
         self.io_ports = array.array("B", [0] * (IO_PORTS))
         self.internal_ram1 = array.array("B", [0] * (INTERNAL_RAM1))
         self.non_io_internal_ram1 = array.array("B", [0] * (NON_IO_INTERNAL_RAM1))
         self.interrupt_register = array.array("B", [0] * (INTERRUPT_ENABLE_REGISTER))
 
+        if randomize: # NOTE: In real life, the RAM is scrambled with random data on boot.
+            for mem in (self.internal_ram0, self.non_io_internal_ram0, self.io_ports, self.internal_ram1,
+                      self.non_io_internal_ram1, self.interrupt_register):
```

Comment:
Only randomize RAM
```suggestion

```

Reply:
What isn't RAM in this case? 

Everything is inside the RAM class, should some things be moved elsewhere?

Reply:
Well, the IO registers and the interrupt register shouldn't be randomized.
I've attached an example from BGB that shows the values of some of the IO Registers.
WRAM, HRAM, VRAM, and OAM do contain random values on boot. Initializing VRAM with random values is redundant though, as the bootrom clears it.
![gbio](https://user-images.githubusercontent.com/55274642/81072652-be076780-8ee6-11ea-9ed8-7c8417adf8e6.png)

Reply:
We still want to randomize VRAM, as we have our own boot ROM, which currently doesn't clear VRAM. If somebody makes a mistake and the VRAM isn't cleared, it will show up.